<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Automated Plotting of Phylogenetic Trees on Spatial Transcriptomics Data • SpatialPhyloPlot</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Automated Plotting of Phylogenetic Trees on Spatial Transcriptomics Data">
<meta name="description" content="An R package to plot phylogenetic trees generated from 10X Visium spatial transcriptomics data in their spatial context.">
<meta property="og:description" content="An R package to plot phylogenetic trees generated from 10X Visium spatial transcriptomics data in their spatial context.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">SpatialPhyloPlot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/exploring_plotting_options_in_SpatialPhyloPlot.html">Exploring plotting options in `SpatialPhyloPlot`</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Wedge-lab/SpatialPhyloPlot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="spatialphyloplot">SpatialPhyloPlot<a class="anchor" aria-label="anchor" href="#spatialphyloplot"></a>
</h1></div>
<p>An R package for plotting phylogenetic trees from Visium 10X spatial transcriptomics data in an automated way.</p>
<p><code>SpatialPhyloPlot</code> is a work in progress - to follow along please watch and star this repository or check back for updates!</p>
<!-- badges: start -->

<p><img src="https://github.com/Wedge-lab/SpatialPhyloPlot/blob/main/man/SpatialPhyloPlot.png?raw=true" align="right" width="250"></p>
<ul>
<li>
<a href="#spatialphyloplot">SpatialPhyloPlot</a>
<ul>
<li><a href="#introduction">Introduction</a></li>
<li><a href="#citation">Citation</a></li>
<li><a href="#installation">Installation</a></li>
<li>
<a href="#example">Example</a>
<ul>
<li><a href="#static-single-sample">Static single sample</a></li>
<li><a href="#static-multi-sample">Static multi-sample</a></li>
<li><a href="#more-detailed-examples">More detailed examples</a></li>
</ul>
</li>
<li><a href="#future-developments">Future Developments</a></li>
</ul>
</li>
</ul>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>SpatialPhyloPlot</code> is an R package for automated plotting of phylogenetic trees derived from spatial transcriptomic data on top of the spatial transcriptomics tissue image. The package is currently compatible with Visium 10X data and phylogenetic trees provided in the Newick file format, such as those produced by MEDICC2.</p>
<p>Please keep an eye out for our upcoming pre-print!</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>If you use <code>SpatialPhyloPlot</code> in your work please cite the package as:</p>
<p>Jakobsdottir M (2025). SpatialPhyloPlot: Automated Plotting of Phylogenetic Trees on Spatial Transcriptomics Data. R package version [insert correct version here], <a href="https://github.com/Wedge-lab/SpatialPhyloPlot" class="external-link uri">https://github.com/Wedge-lab/SpatialPhyloPlot</a>.</p>
<p>A BibTeX entry for LaTeX users is</p>
<pre><code>  @Manual{,
    title = {SpatialPhyloPlot: Automated Plotting of Phylogenetic Trees on Spatial Transcriptomics Data},
    author = {Maria Jakobsdottir},
    year = {2025},
    note = {R package version [insert correct version here]},
    url = {https://github.com/Wedge-lab/SpatialPhyloPlot},
  }</code></pre>
<p>A citation with the version used can be generated by running:</p>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install <code>SpatialPhyloPlot</code> run:</p>
<pre><code><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"Wedge-lab/SpatialPhyloPlot"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p><code>SpatialPhyloPlot</code> has 6 essential inputs:</p>
<ul>
<li>
<code>visium_version</code>: <code>V1</code> or <code>V2</code>
</li>
<li>
<code>newick_file</code>: Path to the <code>.new</code> phylogenetic tree</li>
<li>
<code>image_file</code>: Path to <code>tissue_hires_image.png</code>
</li>
<li>
<code>tissue_positions_file</code>: Path to <code>tissue_positions_list.csv</code>
</li>
<li>
<code>scale_factors</code>: Path to <code>scalefactors_json.json</code>
</li>
<li>
<code>clone_df</code>: A data frame linking Visium barcodes to phylogenetic tree clones</li>
</ul>
<p>We’ll demonstrate how to run <code>SpatialPhyloPlot</code> with some example data provided with the package. The <a href="https://www.10xgenomics.com/datasets/human-brain-cancer-11-mm-capture-area-ffpe-2-standard" class="external-link">Visium 10X data is the Human Brain Cancer 11 mm Capture Area (FFPE) data set provided by 10X Genomics</a>. <strong>Please note: the phylogenetic tree and clones provided are dummy data intended for illustration purposes only and do not represent a formal copy number of phylogenetic tree analysis of this sample.</strong></p>
<pre><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Wedge-lab/SpatialPhyloPlot" class="external-link">SpatialPhyloPlot</a></span><span class="op">)</span></span>
<span><span class="co"># Get directory of files provided with SpatialPhyloPlot</span></span>
<span><span class="va">newick_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"demo.new"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span><span class="va">image_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tissue_hires_image.png"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span><span class="va">tissue_positions_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tissue_positions.csv"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span><span class="va">scale_factors_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scalefactors_json.json"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span><span class="va">clone_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"clone_df.rds"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span><span class="op">)</span></span></code></pre>
<div class="section level3">
<h3 id="static-single-sample">Static single sample<a class="anchor" aria-label="anchor" href="#static-single-sample"></a>
</h3>
<pre><code><span><span class="fu"><a href="reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 clone_barcode_column <span class="op">=</span> <span class="st">"Barcode"</span>,</span>
<span>                 point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                 <span class="op">)</span></span></code></pre>
<p><img src="https://github.com/Wedge-lab/SpatialPhyloPlot/blob/main/man/example_single_sample.png?raw=true" align="center" width="500"></p>
</div>
<div class="section level3">
<h3 id="static-multi-sample">Static multi-sample<a class="anchor" aria-label="anchor" href="#static-multi-sample"></a>
</h3>
<p>If you want to plot a separate phylogenetic tree linking common clones between your two samples we recommend using the <code>plot_connections = TRUE</code>, <code>plot_internal_nodes = TRUE</code>, and <code>shared_clones = TRUE</code> options.</p>
<pre><code><span><span class="fu"><a href="reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 clone_barcode_column <span class="op">=</span> <span class="st">"Barcode"</span>,</span>
<span>                 image_file_left <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 tissue_positions_file_left <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors_left <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 clone_df_left <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 multisample <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 plot_connections <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 plot_internal_nodes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 shared_clones <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 point_alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre>
<p><img src="https://github.com/Wedge-lab/SpatialPhyloPlot/blob/main/man/example_double_sample.png?raw=true" align="center" width="1000"></p>
<p>If you want to plot a single joint phylogenetic tree we recommend using the <code>plot_internal_nodes = TRUE</code> and <code>shared_clones = FALSE</code> options.</p>
<pre><code><span><span class="fu"><a href="reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 clone_barcode_column <span class="op">=</span> <span class="st">"Barcode"</span>,</span>
<span>                 image_file_left <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 tissue_positions_file_left <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors_left <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 clone_df_left <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 multisample <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 plot_internal_nodes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 shared_clones <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                 point_alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre>
<p><img src="https://github.com/Wedge-lab/SpatialPhyloPlot/blob/main/man/example_double_sample_one_tree.png?raw=true" align="center" width="1000"></p>
</div>
<div class="section level3">
<h3 id="more-detailed-examples">More detailed examples<a class="anchor" aria-label="anchor" href="#more-detailed-examples"></a>
</h3>
<p>For more detailed examples that explore different features and customisations that can be achieved through <code>SpatialPhyloPlot</code> please see our <a href="https://wedge-lab.github.io/SpatialPhyloPlot/articles/exploring_plotting_options_in_SpatialPhyloPlot.html">vignette that explores different plotting options</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="future-developments">Future Developments<a class="anchor" aria-label="anchor" href="#future-developments"></a>
</h2>
<p>If you have an idea for a feature that you would like to see implemented in this package please submit a <a href="https://github.com/Wedge-lab/SpatialPhyloPlot/issues" class="external-link">GitHub issue</a> and we’ll see what we can do! :)</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Wedge-lab/SpatialPhyloPlot/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/Wedge-lab/SpatialPhyloPlot" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SpatialPhyloPlot</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Maria Jakobsdottir <br><small class="roles"> Author, maintainer </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://codecov.io/gh/Wedge-lab/SpatialPhyloPlot" class="external-link"><img src="https://codecov.io/gh/Wedge-lab/SpatialPhyloPlot/graph/badge.svg?token=OYOS8HNLTK" alt="codecov"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Maria Jakobsdottir.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
