<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Exploring plotting options in `SpatialPhyloPlot` • SpatialPhyloPlot</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Exploring plotting options in `SpatialPhyloPlot`">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SpatialPhyloPlot</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/exploring_plotting_options_in_SpatialPhyloPlot.html">Exploring plotting options in `SpatialPhyloPlot`</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Wedge-lab/SpatialPhyloPlot/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Exploring plotting options in `SpatialPhyloPlot`</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Wedge-lab/SpatialPhyloPlot/blob/v0.4.0/vignettes/exploring_plotting_options_in_SpatialPhyloPlot.Rmd" class="external-link"><code>vignettes/exploring_plotting_options_in_SpatialPhyloPlot.Rmd</code></a></small>
      <div class="d-none name"><code>exploring_plotting_options_in_SpatialPhyloPlot.Rmd</code></div>
    </div>

    
    
<p>Date compiled: 11 August, 2025.</p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The purpose of this vignette is to show users how to plot an
customise the plotting of phylogenetic trees on top of Visium spatial
transcriptomics data.</p>
<p>Please note, the trees produced by <code>SpatialPhyloPlot</code> are
intended to show the spatial relationship between clones. Thus, the
length of the segments in the tree <strong>does not</strong> represent
genetic distance.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"Wedge-lab/SpatialPhyloPlot"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Wedge-lab/SpatialPhyloPlot" class="external-link">SpatialPhyloPlot</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="inputs">Inputs<a class="anchor" aria-label="anchor" href="#inputs"></a>
</h2>
<p><code>SpatialPhyloPlot</code> has 6 essential inputs:</p>
<ul>
<li>
<code>visium_version</code>: <code>V1</code> or <code>V2</code>
</li>
<li>
<code>newick_file</code>: Path to the <code>.new</code> phylogenetic
tree</li>
<li>
<code>image_file</code>: Path to
<code>tissue_hires_image.png</code>
</li>
<li>
<code>tissue_positions_file</code>: Path to
<code>tissue_positions_list.csv</code>
</li>
<li>
<code>scale_factors</code>: Path to
<code>scalefactors_json.json</code>
</li>
<li>
<code>clone_df</code>: A data frame linking Visium barcodes to
phylogenetic tree clones</li>
</ul>
<p>For the first part of this vignette we’ll be using the <a href="https://www.10xgenomics.com/datasets/human-brain-cancer-11-mm-capture-area-ffpe-2-standard" class="external-link">Visium
10X data is the Human Brain Cancer 11 mm Capture Area (FFPE) data set
provided by 10X Genomics</a> as our Visium data set, with a dummy tree
and some dummy clones for the purposes of illustration.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">newick_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"demo.new"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">image_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tissue_hires_image.png"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tissue_positions_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tissue_positions.csv"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">scale_factors_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scalefactors_json.json"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clone_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"clone_df.rds"</span>, package <span class="op">=</span> <span class="st">"SpatialPhyloPlot"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="basic-plot-features">Basic plot features<a class="anchor" aria-label="anchor" href="#basic-plot-features"></a>
</h2>
<div class="section level3">
<h3 id="visium-spots">Visium spots<a class="anchor" aria-label="anchor" href="#visium-spots"></a>
</h3>
<p>By default, <code>SpatialPhyloPlot</code> plots each Visium spot as a
point with an opacity of 0.8, as <code>plot_points = TRUE</code> and
<code>point_alpha = 0.8</code> by default.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-4-1.png" width="768"></p>
<p>Users can choose to modify the plotting or opacity of points by
modifying these two parameters:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_points <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-5-1.png" width="768"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 point_alpha <span class="op">=</span> <span class="fl">0.2</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-6-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="centroids">Centroids<a class="anchor" aria-label="anchor" href="#centroids"></a>
</h3>
<p>By default, <code>SpatialPhyloPlot</code> plots a central point, or
centroid, for each of the clones, the location of which is calculated as
the average of the x and y coordinates of the spots making up that
clones. These centroids are plotted with a default opacity of 0.9 and a
point size of 8, as <code>centroid_alpha = 0.9</code> and
<code>centroid_size = 8</code> by default. If you do not wish to plot
centroids the easiest way to do this is by setting the
<code>centroid_alpha = 0</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 centroid_alpha <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<p>Similarly, centroid size can be modulated by changing
<code>centroid_size</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 centroid_size <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="polygons">Polygons<a class="anchor" aria-label="anchor" href="#polygons"></a>
</h3>
<p>Polygons can be plotted around each clone by setting
<code>plot_polygon = TRUE</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_polygon <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-9-1.png" width="768"></p>
<p>In order for your polygon to be shaded, you will need to change the
<code>hull_alpha</code> parameter, which is set to 0 by default:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 hull_alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-10-1.png" width="768"></p>
<p>Depending on the nature of your data, you may also want to modulate
the <code>hull_expansion</code> parameter, to customise how tightly your
polygon or hull fits to your points. The default value is 0.005.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 hull_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                 hull_expansion <span class="op">=</span> <span class="fl">0.02</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-11-1.png" width="768"></p>
<p>Plotting polygons without the Visium spots can be a great, minimal,
way of presenting the data in a way that allows the underlying histology
to be visualised as well:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 plot_points <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                 hull_alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-12-1.png" width="768"></p>
<p>Note, plotting polygons (and centroids and trees) on top of your data
may not look as nice if your clones are very intermixed, rather than
spatially distinct:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clone_df_mixed</span> <span class="op">&lt;-</span> <span class="va">clone_df</span></span>
<span><span class="va">clone_df_mixed</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"C"</span>,<span class="st">"D"</span><span class="op">)</span>, <span class="fl">10878</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df_mixed</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_polygon <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 hull_alpha <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="modifying-appearance-of-points">Modifying appearance of points<a class="anchor" aria-label="anchor" href="#modifying-appearance-of-points"></a>
</h3>
<p>The appearance of the points representing Visium spots can be
modified using the <code>point_size</code> and <code>point_shape</code>
parameters.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 point_size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-14-1.png" width="768"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 point_shape <span class="op">=</span> <span class="fl">18</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-15-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="modifying-colours">Modifying colours<a class="anchor" aria-label="anchor" href="#modifying-colours"></a>
</h3>
<div class="section level4">
<h4 id="point-node-and-polygon-colours">Point, node, and polygon colours<a class="anchor" aria-label="anchor" href="#point-node-and-polygon-colours"></a>
</h4>
<p>If you wish to use specific colours to represent your clones, you can
do so by passing a list of colours named after your clones to the
<code>palette</code> parameter.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">my_clone_colours</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"A"</span> <span class="op">=</span> <span class="st">"#183667"</span>,</span>
<span>                         <span class="st">"B"</span> <span class="op">=</span> <span class="st">"#ddebee"</span>,</span>
<span>                         <span class="st">"C"</span> <span class="op">=</span> <span class="st">"#b8badd"</span>,</span>
<span>                         <span class="st">"D"</span> <span class="op">=</span> <span class="st">"#B9D8DC"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">my_clone_colours</span></span>
<span><span class="co">#&gt; $A</span></span>
<span><span class="co">#&gt; [1] "#183667"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $B</span></span>
<span><span class="co">#&gt; [1] "#ddebee"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $C</span></span>
<span><span class="co">#&gt; [1] "#b8badd"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $D</span></span>
<span><span class="co">#&gt; [1] "#B9D8DC"</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 palette <span class="op">=</span> <span class="va">my_clone_colours</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-18-1.png" width="768"></p>
</div>
</div>
<div class="section level3">
<h3 id="modifying-segments">Modifying segments<a class="anchor" aria-label="anchor" href="#modifying-segments"></a>
</h3>
<div class="section level4">
<h4 id="colour">Colour<a class="anchor" aria-label="anchor" href="#colour"></a>
</h4>
<p>The colour of the segments joining the centroids can be changed using
the <code>segment_colour</code> option. The default is
<code>"grey"</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 segment_colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-19-1.png" width="768"></p>
<p>We can also adjust the width and the opacity of the joining segments
by using <code>segment_width</code> and <code>segment_alpha</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 segment_colour <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                 segment_width <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                 segment_alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-20-1.png" width="768"></p>
</div>
</div>
<div class="section level3">
<h3 id="plotting-internal-nodes">Plotting internal nodes<a class="anchor" aria-label="anchor" href="#plotting-internal-nodes"></a>
</h3>
<p>Phylogenetic trees often have internal nodes, that are not
represented by any of the clones in the tissue. If you would like to
plot internal nodes in the tree as separate nodes, this can be done by
setting <code>plot_internal_nodes = TRUE</code>. This is particularly
useful when segments linking clones are close to each other or
overlapping. The position of internal nodes is calculated as the average
x and y position of all nodes connecting to it.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_internal_nodes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-21-1.png" width="768"></p>
<p>The appearance of internal nodes can be adjusted by chaning the
<code>internal_node_colour</code>, <code>internal_node_size</code>, and
<code>internal_node_alpha</code> parameters.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 plot_internal_nodes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 internal_node_colour <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>                 internal_node_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                 internal_node_alpha <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-22-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="anchoring-trees">Anchoring trees<a class="anchor" aria-label="anchor" href="#anchoring-trees"></a>
</h3>
<p>In certain situations, users may with to add an anchor point for
their phylogenetic tree to start from. To do this we need to provide the
coordinates at which we want to plot the anchor point
(<code>origin_coord</code>), and the name of the internal node or clone
that we want the anchor point to refer to, as is appears in the Newick
phylogenetic tree file (<code>origin_name</code>).</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># SpatialPhyloPlot(visium_version = "V2",</span></span>
<span><span class="co">#                  newick_file = newick_file,</span></span>
<span><span class="co">#                  image_file = image_file,</span></span>
<span><span class="co">#                  scale_factors = scale_factors_file,</span></span>
<span><span class="co">#                  tissue_positions_file = tissue_positions_file, </span></span>
<span><span class="co">#                  clone_df = clone_df,</span></span>
<span><span class="co">#                  clone_group_column = "group",</span></span>
<span><span class="co">#                  origin_coord = c(0,0.5),</span></span>
<span><span class="co">#                  origin_name = )</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="plotting-multiple-samples">Plotting multiple samples<a class="anchor" aria-label="anchor" href="#plotting-multiple-samples"></a>
</h2>
<p>Plotting multiple samples is particularly useful when representing
relationships between multiple samples from the same patient, such as
paired primary and metastatic tumours. The first sample that is provided
is always plotted in the middle, in the [0,1] coordinate system. The
other sample, or samples, can be plotted to the left, right, above, or
below the main sample. When plotting multiple samples we assume they
share the same phylogenetic tree and Visium version but we require a
separate image, scale factor, tissue position, and clone data frame.</p>
<p>Below we work through some of the different ways that you could plot
two samples together. When plotting multiple samples together we must
always set <code>multisample = TRUE</code>. Phylogenetic trees can be
plotted separately for each sample, or we can plot a single phylogenetic
tree across both samples. The most appropriate approach will depend on
the underlying data, and how the clones and tree were constructed.</p>
<div class="section level3">
<h3 id="plotting-multiple-samples-with-a-shared-tree">Plotting multiple samples with a shared tree<a class="anchor" aria-label="anchor" href="#plotting-multiple-samples-with-a-shared-tree"></a>
</h3>
<p>To plot a single tree across the two samples, we simply need to
specify that we are running <code>multisample = TRUE</code> and provide
the necessary input files. This is particularly useful when copy number
has been inferred separately for different samples but the phylogenetic
tree has been built jointly.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 multisample <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 image_file_left <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 clone_df_left <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 tissue_positions_file_left <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors_left <span class="op">=</span> <span class="va">scale_factors_file</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-24-1.png" width="1536"></p>
</div>
<div class="section level3">
<h3 id="plotting-multiple-samples-with-separate-trees">Plotting multiple samples with separate trees<a class="anchor" aria-label="anchor" href="#plotting-multiple-samples-with-separate-trees"></a>
</h3>
<p>In other cases it may be more appropriate to plot a separate tree for
each sample, in particular if we have clones that are shared across
samples which can occur when copy number has been inferred jointly. In
this case we specify <code>shared_clones = TRUE</code> to indicate that
one or more of the same clone are found in both samples.
<code>SpatialPhyloPlot</code> will then plot the relevant portion of the
tree for each.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 multisample <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 image_file_left <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 clone_df_left <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 tissue_positions_file_left <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors_left <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 shared_clones <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-25-1.png" width="1536"></p>
<p>When plotting cases where only a couple of clones are shared between
the two samples, it can be useful to highlight that these are the same
by plotting dashed connections between them setting
<code>plot_connections = TRUE</code>. The appearance of these
connections can be adjusted by altering the
<code>connection_width</code> and <code>connection_colour</code>
parameters.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                 multisample <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 image_file_left <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 clone_df_left <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 tissue_positions_file_left <span class="op">=</span> <span class="va">tissue_positions_file</span>,</span>
<span>                 scale_factors_left <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 shared_clones <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 plot_connections <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-26-1.png" width="1536"></p>
</div>
</div>
<div class="section level2">
<h2 id="adding-annotations">Adding annotations<a class="anchor" aria-label="anchor" href="#adding-annotations"></a>
</h2>
<p>As the outputs of <code>SpatialPhyloPlot</code> are
<code>ggplot2</code> plots they can easily be annotated and customised
using <code>ggplot2</code> functions. The [0,1] coordinate system that
each plot uses makes placing annotations simple as well. Annotating
plots can be particularly useful when highlighting areas with
interesting pathology.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/SpatialPhyloPlot.html">SpatialPhyloPlot</a></span><span class="op">(</span>visium_version <span class="op">=</span> <span class="st">"V2"</span>,</span>
<span>                 newick_file <span class="op">=</span> <span class="va">newick_file</span>,</span>
<span>                 image_file <span class="op">=</span> <span class="va">image_file</span>,</span>
<span>                 scale_factors <span class="op">=</span> <span class="va">scale_factors_file</span>,</span>
<span>                 tissue_positions_file <span class="op">=</span> <span class="va">tissue_positions_file</span>, </span>
<span>                 clone_df <span class="op">=</span> <span class="va">clone_df</span>,</span>
<span>                 clone_group_column <span class="op">=</span> <span class="st">"group"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/annotate.html" class="external-link">annotate</a></span><span class="op">(</span><span class="st">"text"</span>, x <span class="op">=</span> <span class="fl">0.8</span>, y <span class="op">=</span> <span class="fl">0.8</span>, label <span class="op">=</span> <span class="st">"Really interesting\narea"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html" class="external-link">geom_rect</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>xmin <span class="op">=</span> <span class="fl">0.7</span>, xmax <span class="op">=</span> <span class="fl">0.9</span>, ymin <span class="op">=</span> <span class="fl">0.5</span>, ymax <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span>, fill <span class="op">=</span> <span class="cn">NA</span>, colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="exploring_plotting_options_in_SpatialPhyloPlot_files/figure-html/unnamed-chunk-28-1.png" width="768"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Maria Jakobsdottir.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
